stackslib/src/chainstate/nakamoto/tests/mod.rs

```diff
@@ -1636,3 +1643,272 @@ pub fn test_get_highest_nakamoto_tenure() {
+            canonical_stacks_tip_consensus_hash: ConsensusHash([((i + 1) as u8); 20]),
+            miner_pk_hash: None,
+        };
+        let winning_block_commit = LeaderBlockCommitOp {
+            sunset_burn: 0,
+            block_header_hash: BlockHeaderHash([(i as u8); 32]),
+            new_seed: VRFSeed([(i as u8); 32]),
+            parent_block_ptr: last_snapshot.block_height as u32,
+            parent_vtxindex: 1,
+            // miners take turns winning
+            key_block_ptr: miners[i - 3].block_height as u32,
+            key_vtxindex: miners[i - 3].vtxindex as u16,
+            memo: vec![STACKS_EPOCH_3_0_MARKER],
+            commit_outs: vec![],
+
+            burn_fee: 12345,
+            input: (Txid([0; 32]), 0),
+            apparent_sender: BurnchainSigner::mock_parts(
+                AddressHashMode::SerializeP2PKH,
+                1,
+                vec![StacksPublicKey::from_hex(
+                    "02d8015134d9db8178ac93acbc43170a2f20febba5087a5b0437058765ad5133d0",
+                )
+                .unwrap()],
+            ),
+
+            txid: winning_txid.clone(),
+            vtxindex: 1,
+            block_height: snapshot.block_height,
+            burn_parent_modulus: ((snapshot.block_height - 1) % BURN_BLOCK_MINED_AT_MODULUS) as u8,
+            burn_header_hash: snapshot.burn_header_hash.clone(),
+        };
+
+        let winning_ops = if i == 3 {
+            // first snapshot includes leader keys
+            miners
+                .clone()
+                .into_iter()
+                .map(|miner| BlockstackOperationType::LeaderKeyRegister(miner))
+                .collect()
+        } else {
+            // subsequent ones include block-commits
+            if sortition {
+                vec![BlockstackOperationType::LeaderBlockCommit(
+                    winning_block_commit,
+                )]
+            } else {
+                vec![]
+            }
+        };
+
+        {
+            let mut tx = SortitionHandleTx::begin(sort_db, &last_snapshot.sortition_id).unwrap();
+            let _index_root = tx
+                .append_chain_tip_snapshot(
+                    &last_snapshot,
+                    &snapshot,
+                    &winning_ops,
+                    &vec![],
+                    None,
+                    None,
+                    None,
+                )
+                .unwrap();
+            tx.test_update_canonical_stacks_tip(
+                &snapshot.sortition_id,
+                &snapshot.canonical_stacks_tip_consensus_hash,
+                &snapshot.canonical_stacks_tip_hash,
+                snapshot.canonical_stacks_tip_height,
+            )
+            .unwrap();
+            tx.commit().unwrap();
+        }
+
+        last_snapshot = SortitionDB::get_block_snapshot(sort_db.conn(), &snapshot.sortition_id)
+            .unwrap()
+            .unwrap();
+
+        if i > 3 {
```

Comment:
What happens if `make_miners_stackerdb_config()` is invoked at `i == 3` ?

Reply:
It would panic because there would be no block-commit associated with the first-ever snapshot (which is what the code would return when asked for the last snapshot with a sortition, when there are no associated block-commits).

Reply:
It seems like it would return an error of some kind from `get_last_snapshot_with_sortition` rather than panic. If that's the case, the test should check that error case.

Reply:
It doesn't panic. But it does generate a result that I am having trouble figuring out what it means XD the assigned miner private key for the stacker dbs writers is not one I can seem to figure out where it comes from